src/main/java/de/embl/cba/mobie/bookmark/BookmarksManager.java

```diff
@@ -75,6 +91,131 @@ public void adaptViewerTransform( Bookmark bookmark )
 		}
 	}
 
+	public void loadAdditionalBookmarks() {
+			Map<String, Bookmark> additionalBookmarks = bookmarksJsonParser.selectAndLoadBookmarks();
+			if (additionalBookmarks != null) {
+				nameToBookmark.putAll(additionalBookmarks);
+				bookmarkDropDown.removeAllItems();
+				for (String bookmarkName : nameToBookmark.keySet()) {
+					bookmarkDropDown.addItem(bookmarkName);
+				}
+			}
+	}
+
+	public void saveCurrentSettingsAsBookmark () {
+		ArrayList<String> bookmarkNameAndLocation = bookmarkSaveDialog();
+		Bookmark currentBookmark = getBookmarkFromCurrentSettings(bookmarkNameAndLocation.get(0));
+		ArrayList<Bookmark> bookmarks = new ArrayList<>();
+		bookmarks.add(currentBookmark);
+
+		if (bookmarkNameAndLocation.get(1).equals(FileUtils.PROJECT) &&
+				bookmarksJsonParser.getDatasetLocation().contains( "raw.githubusercontent" )) {
+			bookmarksJsonParser.saveBookmarksToGithub(bookmarks);
+		} else {
+			try {
+				bookmarksJsonParser.saveBookmarksToFile(bookmarks, bookmarkNameAndLocation.get(1));
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+	private ArrayList<String> bookmarkSaveDialog () {
+		String fileLocation = null;
+		String bookmarkName = null;
+		final GenericDialog gd = new GenericDialog( "Choose save location" );
+		gd.addStringField("Bookmark Name", "name");
+		gd.addChoice( "Save to", new String[]{FileUtils.PROJECT, FileUtils.FILE_SYSTEM }, FileUtils.PROJECT );
+		gd.showDialog();
+		if ( gd.wasCanceled() ) return null;
+		bookmarkName = gd.getNextString();
+		fileLocation = gd.getNextChoice();
+
+		ArrayList<String> bookmarkNameandLocation = new ArrayList<>();
+		bookmarkNameandLocation.add(bookmarkName);
+		bookmarkNameandLocation.add(fileLocation);
+		return bookmarkNameandLocation;
+	}
+
+	public Bookmark getBookmarkFromCurrentSettings( String bookmarkName) {
+		HashMap< String, MutableImageProperties > layers = new HashMap<>();
+		Set<String> visibleSourceNames = sourcesPanel.getVisibleSourceNames();
+
+		for (String sourceName : visibleSourceNames) {
+			MutableImageProperties sourceImageProperties = getCurrentImageProperties(sourceName);
+			layers.put(sourceName, sourceImageProperties);
+		}
+
+		BdvHandle bdv = sourcesPanel.getBdv();
+		Bookmark currentBookmark = new Bookmark();
+		currentBookmark.name = bookmarkName;
+		currentBookmark.layers = layers;
+		double[] currentPosition = new double[3];
+		BdvUtils.getGlobalMouseCoordinates(bdv).localize(currentPosition);
+		currentBookmark.position = currentPosition;
+		currentBookmark.normView = Utils.createNormalisedViewerTransformString( bdv, Utils.getMousePosition( bdv ) ).split(",");
+		currentBookmark.view = null;
+
+		return currentBookmark;
+	}
+
+	private MutableImageProperties getCurrentImageProperties(String sourceName) {
```

Comment:
I would rename to 

`fetchMutableSourceProperties( String sourceName )`

`fetch` being different from `get` and `create` in a sense that it does no really `create` something new but just looks around and `fetches` stuff ;-)
but whatever... :-)